### PR TITLE
expr2c now distinguishes binary and multi-ary expressions

### PR DESCRIPTION
**Pull request by kroening**
multi-ary and proper binary expressions can now have different textual representations; among other benefits, this allows considering associativity properly.

### CODE REVIEW

**Comment by tautschnig** [comment]
Would you mind copying the PR description into the commit message's body?

File: src/ansi-c/expr2c.cpp [hunk 1]
```diff
@@ -1028,6 +1027,57 @@ std::string expr2ct::convert_binary(
   const std::string &symbol,
   unsigned precedence,
   bool full_parentheses)
+{
+  if(src.operands().size()!=2)
+    return convert_norep(src, precedence);
```
Comment (tautschnig) [inline src/ansi-c/expr2c.cpp:1032]: Shouldn't that now be a `PRECONDITON`?

**Review by peterschrammel — APPROVED** [review]
LGTM apart from Michael's precondition comment, which I agree with.

File: src/ansi-c/expr2c.cpp (hunk 1 above)
Comment (tautschnig) [reply src/ansi-c/expr2c.cpp:1032]: @kroening Would you mind fixing/commenting on this?

Comment (kroening) [reply src/ansi-c/expr2c.cpp:1032]: expr2c has, historically, not enforced any data invariants on expressions; malformed expressions are instead printed as "Lisp" expressions.

Using a PRECONDITION here would be a very substantial change in semantics.

**Review by tautschnig — APPROVED** [review]
Approving with #1760 opened for discussion.

**Pull request merged**